### PR TITLE
refactor: abstracting the creation of a vector store

### PR DESCRIPTION
**Pull request by SEBRATHEZEBRA**

### CODE REVIEW

File: src/common/model/createMemoryStore.test.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import { MemoryVectorStore } from 'langchain/vectorstores/memory';
+import { CreateMemoryStore } from './createMemoryStore';
+import { OpenAIEmbeddings } from 'langchain/embeddings/openai';
+
+describe('CreateMemoryStore function', () => {
+  const initialFiles = [
```
Comment (mattzcarey) [inline src/common/model/createMemoryStore.test.ts:6]: can you make this file in the testFiles and then read it here. Makes the code a lot nicer.

File: src/common/model/createMemoryStore.test.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+  it('Checks if the function provides the required functionality', async () => {
+    const result = await CreateMemoryStore(initialFiles);
+
+    const expectedResult = await MemoryVectorStore.fromDocuments(
+      initialFiles,
+      new OpenAIEmbeddings(),
+      {}
+    );
+    expect(result).toEqual(expectedResult);
```
Comment (mattzcarey) [inline src/common/model/createMemoryStore.test.ts:70]: also can you check that a simularitySearchWithScore returns a number. Thats the most used method on this class.

File: src/common/model/createMemoryStore.test.ts (hunk 1 above)
Comment (SEBRATHEZEBRA) [reply src/common/model/createMemoryStore.test.ts]: done :)

File: src/common/model/createMemoryStore.test.ts (hunk 2 above)
Comment (SEBRATHEZEBRA) [reply src/common/model/createMemoryStore.test.ts:21]: done :)

**Review by mattzcarey — APPROVED** [review]

**Pull request merged**